docs/trace/README.md

```diff
@@ -1,19 +1,122 @@
+traces with OpenTelemetry .NET:
+
+* [Getting Started - ASP.NET Core
+  Application](./getting-started-aspnetcore/README.md)
+* [Getting Started - Console Application](./getting-started-console/README.md)
+
+## Package Version
+
+:heavy_check_mark: You should always use the
+[System.Diagnostics.Activity](https://learn.microsoft.com/dotnet/api/system.diagnostics.activity)
+APIs from the latest stable version of
+[System.Diagnostics.DiagnosticSource](https://www.nuget.org/packages/System.Diagnostics.DiagnosticSource/)
+package, regardless of the .NET runtime version being used:
+
+* If you're using the latest stable version of [OpenTelemetry .NET
+  SDK](../../src/OpenTelemetry/README.md), you don't have to worry about the
+  version of `System.Diagnostics.DiagnosticSource` package because it is already
+  taken care of for you via [package
+  dependency](../../Directory.Packages.props).
+* The .NET runtime team is holding a high bar for backward compatibility on
+  `System.Diagnostics.DiagnosticSource` even during major version bumps, so
+  compatibility is not a concern here.
+
+## Tracing API
+
+:stop_sign: You should avoid creating
+[`ActivitySource`](https://learn.microsoft.com/dotnet/api/system.diagnostics.activitysource)
+too frequently. `ActivitySource` is fairly expensive and meant to be reused
+throughout the application. For most applications, it can be modeled as static
+readonly field (e.g. [Program.cs](./getting-started-console/Program.cs)) or
+singleton via dependency injection (e.g.
+[Instrumentation.cs](../../examples/AspNetCore/Instrumentation.cs)).
+
+:heavy_check_mark: You should use dot-separated
+[UpperCamelCase](https://en.wikipedia.org/wiki/Camel_case) as the
+[`ActivitySource.Name`](https://learn.microsoft.com/dotnet/api/system.diagnostics.activitysource.name).
+In many cases, using the fully qualified class name might be a good option.
+
+```csharp
+static readonly ActivitySource MyActivitySource = new("MyCompany.MyProduct.MyLibrary");
+```
+
+:heavy_check_mark: You should check
+[`Activity.IsAllDataRequested`](https://learn.microsoft.com/dotnet/api/system.diagnostics.activity.isalldatarequested)
+before [setting
+Tags](https://learn.microsoft.com/dotnet/api/system.diagnostics.activity.settag)
+for better performance.
+
+```csharp
+using (var activity = MyActivitySource.StartActivity("SayHello"))
+{
+    if (activity != null && activity.IsAllDataRequested == true)
+    {
+        activity.SetTag("http.url", "http://www.mywebsite.com");
+    }
+}
+```
+
+:heavy_check_mark: You should use
+[Activity.SetTag](https://learn.microsoft.com/dotnet/api/system.diagnostics.activity.settag)
+instead of
+[Activity.AddTag](https://learn.microsoft.com/dotnet/api/system.diagnostics.activity.addtag)
+and
+[Activity.SetCustomProperty](https://learn.microsoft.com/dotnet/api/system.diagnostics.activity.setcustomproperty)
+because the latter do not perform deduplication.
```

Comment:
https://github.com/open-telemetry/opentelemetry-dotnet/blob/main/test/Benchmarks/Trace/ActivityCreationBenchmarks.cs#L22-L23 There is some advantage is not doing de-dedup. Wondering if that should be even mentioned here? If the user is in full control on tag keys, using AddTag can save cpu cycles.

Reply:
I feel it is diluting the message + hard to maintain (nobody should assume that the activity instances are solely managed by them since anyone can build processors).